Bottom: 7559859822bb1aea47b842262917d133dbcbb47d
Top:    8470ff5f70eade3360f14aa54f1a5c31cb638a36
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-03-22 20:09:29 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 31021c9..b01a5ae 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1101,7 +1101,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
